api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -448,4 +448,52 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+	
+	/**
+	 * @see VisitValidator#validate(Object,Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfVisitDateBeforeBirthDate() {
+		VisitService vs = Context.getVisitService();
+		Visit visit = new Visit();
+		
+		visit.setVisitType(vs.getVisitType(1));
+		visit.setPatient(Context.getPatientService().getPatient(2));
+		Calendar c = Calendar.getInstance();
+		c.set(1974, 4, 8);
+		visit.setStartDatetime(c.getTime());
+		
+		Errors errors = new BindException(visit, "visit");
+		new VisitValidator().validate(visit, errors);
+		assertTrue(errors.hasFieldErrors("startDatetime"));
+	}
+	
+	/**
+	 * @see VisitValidator#validate(Object,Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfVisitDateBeforeEstimatedBirthDate() {
+		VisitService vs = Context.getVisitService();
```

Comment:
please use the `visitService` thats already a field of the test class and set up in the `@Before`

Reply:
When I refactored this, I actually ditched using a visitService at all and instead am calling the method "private Visit makeVisit(Integer patientId)". Hopefully that makes sense.